hack/format-owners-aliases.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pmezard/go-difflib/difflib"
+	yaml "gopkg.in/yaml.v3"
+)
+
+const (
+	aliasesPrefix = "aliases:\n"
+	customStart   = "## BEGIN CUSTOM CONTENT"
+	customEnd     = "## END CUSTOM CONTENT"
+)
+
+type flags struct {
+	diff   bool
+	write  bool
+	custom bool
+}
+
+type aliases struct {
+	Aliases map[string][]string
+}
+
+func removeDetectDupes(s []string, detect bool) ([]string, error) {
+	seen := make(map[string]struct{}, len(s))
+	i := 0
+	for _, v := range s {
+		if k, ok := seen[v]; ok {
+			if detect {
+				return nil, errors.Errorf("list %v contains duplicate element %q", s, k)
+			}
+			continue
+		}
+		seen[v] = struct{}{}
+		s[i] = v
+		i++
+	}
+	return s[:i], nil
+}
+
+func (a *aliases) marshalYAML() ([]byte, error) {
+	// append the keys to a string slice
+	keys := make(sort.StringSlice, len(a.Aliases))
+	i := 0
+	for k := range a.Aliases {
+		keys[i] = k
+		// remove dupes and sort the list of OWNERS
+		a.Aliases[k], _ = removeDetectDupes(a.Aliases[k], false)
+		sort.Strings(a.Aliases[k])
+		i++
+	}
+	// error out on duplicate keys (group names), otherwise sort the keys
+	keys, err := removeDetectDupes(keys, true)
```

Comment:
the better alternative here might be auto-merge duplicate groups:
```
A:
  - A
  - B
A:
  - A
  - B
  - C
```
to result in:
```
A:
  - A
  - B
  - C
```

Reply:
actually, this is too entangled because the source is a `map` with multiple keys.